### PR TITLE
documentation: More work on documentation Fix #86

### LINKED ISSUE #86: More work on documentation (reset)
_Migrated from JIRA [RUCIO-2861](https://its.cern.ch/jira/browse/RUCIO-2861)_

### PR DESCRIPTION
**Pull request by vingar**

### CODE REVIEW

File: doc/source/Architecture.rst [hunk 1]
```diff
@@ -1,3 +1,24 @@
...
 Architecture
-------------
+============
+
+ the Rucio system is based on a distributed architecture and can be decomposed into four
+ main components: Server, Daemons, Resources, and User Interface.
+
+The Rucio server is a passive component listening to incoming queries. It connects several
+Rucio core components together and offers a common interface for external interaction.
+This abstraction layer is responsible for the interactions with different
```
Comment (bari12) [inline doc/source/Architecture.rst:9]: I think with "This Abstraction layer" RSEs are being ment? There is something missing here.

File: doc/source/Architecture.rst [hunk 2]
```diff
@@ -1,3 +1,24 @@
...
+middleware tools and storage systems. It effectively hides the complexity of these tools
+and combines them into one interface used by Rucio.  The persistence layer keeps all the
```
Comment (bari12) [inline doc/source/Architecture.rst:11]: Persistence layer also seems to suddenly appear without much reason.

File: doc/source/conf.py [hunk 3]
```diff
@@ -1,230 +1,173 @@
...
 master_doc = 'index'
 
 # General information about the project.
 project = u'Rucio'
-copyright = u'2015, CERN'
-
-# Show ToDos
-todo_include_todos = True
+copyright = u'2017, rucio-dev@googlegroups.com'
+author = u'rucio-dev@googlegroups.com'
```
Comment (bari12) [inline doc/source/conf.py]: rucio-dev@cern.ch

File: doc/source/index.rst [hunk 4]
```diff
@@ -1,145 +1,63 @@
...
-detector as well as generated or derived, in the ATLAS distributed computing system.
-Rucio uses to manage accounts, files, datasets and distibuted storage systems.
+Rucio is a project that provides services and associated libraries for allowing scientific
+collaborations to manage large volumes of data spread across facilities at
+multiple institutions and organisations. Rucio has been developed by
+`the ATLAS experiment <https://atlas.cern/>`_.
+
+Rucio offers advanced features, is highly scalable and modular. It is a data management
+solution that could cover the needs of different communities in the scientific
+domain (e.g., HEP, astronomy, biology).
```
Comment (bari12) [inline doc/source/index.rst:16]: Again, here we should put the final text once checked.

File: doc/source/monitoring.rst [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+ * Rucio internal monitoring using Graphite/Grafana
+ * Transfer monitoring using the messages sent by Hermes
+ * File/Dataset Access monitoring using the traces
+
+
+
+Internal Monitoring:
+--------------------
+
+This is to monitor the internals of Rucio server and daemons, e.g., submission rate of the conveyor, state of conveyor queues, reaper deletion rate, server response times, server active session, etc. We use Graphite[1] for this. It’s easy to setup and then you just have to point your Rucio instance to the Graphite server using the “carbon_server” options in the “monitor” section in etc/rucio.cfg. The different Rucio components then will send metrics using those “record_*” functions you will find all over the code. Graphite has a built-in web interface to show graphs but more comfortable to use is Grafana. You can find an example dashboard from us here [2].
```
Comment (bari12) [inline doc/source/monitoring.rst]: - server response times -> server response time
- server active session -> server active sessions
- I would remove sentences like "It's easy to setup"...

File: doc/source/monitoring.rst [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+
+
+Transfer monitoring:
+-------------------_
+If a transfer is submitted, queued, waiting, done or failed messages are sent out using Hermes to ActiveMQ and are also archived in the messages_history table. Same is true for deletions. In our case we have a dedicated monitoring infrastructure by the CERN IT monitoring group, that reads the messages from ActiveMQ, aggregates them and then writes the aggregated data into Elasticsearch/InfluxDB from where it then can be visualised using Kibana/Grafana. You told me that you already have you own Elasticsearch running. So, I think for your use case and scale it would be enough if you would take message from ActiveMQ and write them directly to ES using Logstash [3]. From there it should then be easy to create plots.
```
Comment (bari12) [inline doc/source/monitoring.rst]: - failed -> failed,
- messages are sent out ... -> messages are sent to ActiveMQ via Hermes. 
- In our case -> In ATLAS... or something like that
- You told me .... -> ditch
- So I think for you ... -> ditch

File: doc/source/monitoring.rst [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+
+
+Access monitoring:
+------------------
+
+And last there are the traces. Those are sent by the pilots or the rucio clients whenever a file is downloaded/uploaded. We use them on one hand to update the “accessed_at” column on the DIDS, COLLECTION_REPLICAS, DATASET_LOCKS and REPLICAS table and on the other hand to create popularity reports which then can be used to check which datasetd have been used a lot or not at all. For this to work you would need to setup a tracer server. We have separate machines for that but it can also be on your normal Rucio server. The clients/pilots are then sending traces there using HTTP POST. The server then has to be configured in the “trace” section in etc/rucio.cfg to send to your ActiveMQ server. From there you can then again write the traces to Elasticsearch. To update the “accessed_at” column we have a specific daemon called Kronos. This daemon read the traces from ActiveMQ and then updates the corresponding DB tables.
```
Comment (bari12) [inline doc/source/monitoring.rst]: - And last there are the traces -> ditch
- Those -> Traces
- Again the rest of the paragraph seems to describe "We" as ATLAS.

**Review by bari12 — CHANGES_REQUESTED** [review]

File: doc/source/monitoring.rst (hunk 5 above)
Comment (vingar) [reply doc/source/monitoring.rst]: for reference: original text is from Thomas' mail

File: doc/source/monitoring.rst [hunk 8]
```diff
@@ -0,0 +1,34 @@
+Monitoring
+===========
+
+There are mainly three different things:
```
Comment (mlassnig) [inline doc/source/monitoring.rst:4]: There are three different monitoring components:

File: doc/source/monitoring.rst [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+ * Rucio internal monitoring using Graphite/Grafana
+ * Transfer monitoring using the messages sent by Hermes
+ * File/Dataset Access monitoring using the traces
+
+
+Internal Monitoring:
+--------------------
+
+This is to monitor the internals of Rucio server and
```
Comment (mlassnig) [inline doc/source/monitoring.rst:13]: of the Rucio servers

File: doc/source/monitoring.rst [hunk 10]
```diff
@@ -0,0 +1,34 @@
...
+daemons, e.g., submission rate of the conveyor, state of conveyor queues, reaper
+deletion rate, server response times, server active session, etc. We use Graphite[1]
+for this. It’s easy to setup and then you just have to point your Rucio instance to the
```
Comment (mlassnig) [inline doc/source/monitoring.rst:16]: remove "just" ;-)

File: doc/source/monitoring.rst [hunk 11]
```diff
@@ -0,0 +1,34 @@
...
+Graphite server using the “carbon_server” options in the “monitor” section in etc/rucio.cfg.
+The different Rucio components then will send metrics using those “record_*” functions you
+ will find all over the code. Graphite has a built-in web interface to show graphs but more comfortable to use is Grafana. You can find an example dashboard from us here [2].
+
+
+Transfer monitoring:
+-------------------_
+
+If a transfer is submitted, queued, waiting, done or failed  messages are sent to ActiveMQ via Hermes and are also archived in the messages_history table. Same is true for deletions.
+In the ATLAS, case we have a dedicated monitoring infrastructure,
```
Comment (mlassnig) [inline doc/source/monitoring.rst:26]: In the ATLAS case

File: doc/source/monitoring.rst [hunk 12]
```diff
@@ -0,0 +1,34 @@
...
+that reads the messages from ActiveMQ, aggregates them and then writes the aggregated data
```
Comment (mlassnig) [inline doc/source/monitoring.rst:27]: no comma before "that" (commas only before "which") ;-)

File: doc/source/index.rst [hunk 13]
```diff
@@ -1,145 +1,63 @@
...
-    overview_Replica_management
-    replication_rules_examples
-    RSE_Expressions
-    overview_Accounting_and_quota
-    overview_Notifications
-    overview_Subscriptions
-
-Architecture
-============
+This section sontains the general information related to Rucio which is commom to
```
Comment (mlassnig) [inline doc/source/index.rst:25]: contains
common

File: doc/source/index.rst (hunk 4 above)
Comment (mlassnig) [reply doc/source/index.rst:16]: The last sentence seems a bit like advertisement. It's fine for now, but we should revise it later.

File: doc/source/clients.rst [hunk 14]
```diff
@@ -0,0 +1,5 @@
+Rucio Clients
+=============
+
+Rucio includes a client class to removes some of the complexity of dealing with raw HTTP
```
Comment (mlassnig) [inline doc/source/clients.rst:4]: remove

File: doc/source/Architecture.rst [hunk 15]
```diff
@@ -1,3 +1,24 @@
...
+logical data and the application states.
+
+The Rucio daemons are active components that orchestrate the collaborative work of the
+ whole system. They are for example:
+ * Conveyor -  in charge of File transfers
```
Comment (mlassnig) [inline doc/source/Architecture.rst:16]: I think we should flip this around to focus on the functionality, and not the name of the daemons

* Transfer daemon (Conveyor)
* Data expiration (Undertaker)
* File deletion (Reaper)
...
etc.

**Pull request closed without merging**